plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestingThriftHttpMetastoreServer.java

```diff
@@ -77,7 +77,10 @@ private static <T> T proxyHandler(ThriftMetastore delegate, Class<T> iface)
         return newProxy(iface, (proxy, method, args) -> switch (method.getName()) {
             case "getAllDatabases" -> delegate.getAllDatabases();
             case "getDatabase" -> {
-                Optional<Database> optionalDatabase = delegate.getDatabase(args[0].toString());
+                String catalogDatabaseName = args[0].toString();
+                String dbName = catalogDatabaseName != null && !catalogDatabaseName.isEmpty() && catalogDatabaseName.startsWith("@") && catalogDatabaseName.indexOf("#") > 1 ?
+                        catalogDatabaseName.substring(catalogDatabaseName.indexOf("#") + 1) : catalogDatabaseName;
+                Optional<Database> optionalDatabase = delegate.getDatabase(dbName);
```

Comment:
Do we still need this logic when the catalogName is known to be empty?

Reply:
yes, we can remove this.